scripts/question_answering/model.py

```diff
@@ -0,0 +1,752 @@
+
+    Parameters
+    ----------
+    axis : int, default -1
+        The axis to sum over when computing softmax and entropy.
+    sparse_label : bool, default True
+        Whether label is an integer array instead of probalbility distribution.
+    from_logits : bool, default False
+        Whether input is a log probability (usually from log_softmax) instead of
+        unnormalized numbers.
+    weight : float or None
+        Global scalar weight for loss.
+    batch_axis : int, default 0
+        The axis that represents mini-batch.
+    """
+
+    def __init__(self, axis=-1, sparse_label=True, from_logits=False, weight=None, batch_axis=0,
+                 **kwargs):
+        super(MySoftmaxCrossEntropy, self).__init__(
+            weight, batch_axis, **kwargs)
+        self.loss = gluon.loss.SoftmaxCrossEntropyLoss(axis=axis,
+                                                       sparse_label=sparse_label,
+                                                       from_logits=from_logits,
+                                                       weight=weight,
+                                                       batch_axis=batch_axis)
+
+    def forward(self, predict_begin, predict_end, label_begin, label_end):
+        r"""Implement forward computation.
+
+        Parameters
+        -----------
+        predict_begin : NDArray
+            Predicted probability distribution of answer begin position,
+            input tensor with shape `(batch_size, sequence_length)`
+        predict_end : NDArray
+            Predicted probability distribution of answer end position,
+            input tensor with shape `(batch_size, sequence_length)`
+        label_begin : NDArray
+            True label of the answer begin position,
+            input tensor with shape `(batch_size, )`
+        label_end : NDArray
+            True label of the answer end position,
+            input tensor with shape `(batch_size, )`
+
+        Returns
+        --------
+        out: NDArray
+            output tensor with shape `(batch_size, )`
+        """
+        return self.loss(predict_begin, label_begin) + self.loss(predict_end, label_end)
+
+
+class QANet(gluon.HybridBlock):
+    r"""QANet model.
+    We implemented the QANet proposed in the following work::
+        @article{DBLP:journals/corr/abs-1804-09541,
+            author    = {Adams Wei Yu and
+                        David Dohan and
+                        Minh{-}Thang Luong and
+                        Rui Zhao and
+                        Kai Chen and
+                        Mohammad Norouzi and
+                        Quoc V. Le},
+            title     = {QANet: Combining Local Convolution with Global Self-Attention for
+                        Reading Comprehension},
+            year      = {2018},
+            url       = {http://arxiv.org/abs/1804.09541}
+        }
+
+    """
+
+    def __init__(self, **kwargs):

```

Comment:
Please add keyword arguments for the options that you currently retrieve from `opt`, e.g. `opt.char_emb_dim`. The reason is, that if you require `opt`, QANet can only be used when people use the exact same `argparse` setup as in this script.